homeassistant/components/media_player/yamaha.py

```diff
@@ -149,6 +149,11 @@ def __init__(self, name, receiver, source_ignore,
         self._name = name
         self._zone = receiver.zone
 
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return '{0}:{1}'.format(self.receiver.ctrl_url, self._zone)
+        
```

Comment:
Is `self._zone` a user-configurable string (i.e., something that could change through, say, the Yamaha app)? The objective is to not have anything user-configurable in the unique ID that could change down the road while still representing the same thing.

Reply:
It does not appear to, it appears as if it comes from rxv.RXV() call which returns a receiver object.  After tracing through the rxv code, it appears to grab the self._zone from the xml that is requested from the device.  When I get home, I can verify this.

Reply:
I'll have to check the yamaha app as well.

Reply:
I'm guessing the reason it was removed was that it's based on the URL. A unique ID needs to be based on something that won't change for the device, such as a serial number or mac address. I'd see if something like that is available from the API that could be used instead.

Reply:
That makes sense.  For time constraints, I only 'rolled back' the changes.  When i get home I can find a better way to handle the unique_id.  I may have been too hasty with this PR.

Reply:
self._zone is not user configurable, it comes from the device.

As far as I can see in both the rxv code, and poking at the XML API for the Yamaha unit, there is nothing available at the rxv layer for uniqueness. That's all operating at the url level (which there was already code to ignore by url devices previously discovered).

The only existing path I'm seeing to get an actual unique id is data that's available in the upnp discovery packet. We don't require everyone to go through that path, so that wouldn't work if discovery was disabled. I don't see any imperative path to get that.

Newer receivers have a more REST like API, but for those of us with older ones losing multizone because there is not a guarunteed way to generate stable uuid would be really sad.

Reply:
I'd try and investigate the underlying bug. Not having a unique ID shouldn't affect the behavior of the actual platform code at all.